packages/ember-data/lib/system/relationship-meta.js

```diff
@@ -16,13 +16,24 @@ export function typeForRelationshipMeta(store, meta) {
   return type;
 }
 
+export function generateLink(store, kind, type) {
```

Comment:
I may be misunderstanding the use but I was thinking If i put here the function would be looked up only once when the relationship is defined rater than each time the record is updated.